app/models/miq_report/generator.rb

```diff
@@ -164,8 +164,8 @@ def queue_generate_table(options = {})
     if sync
       _async_generate_table(task.id, options)
     else
-      MiqQueue.put(
-        :queue_name  => "reporting",
+      MiqQueue.submit_job(
+        :service     => "reporting",
```

Comment:
To be clear, `submit_job` will set the `role` as `reporting` and because of this, `determine_queue_zone`, from `put`, will set the zone to `nil` for "any" zone.